In [9]:
import numpy as np
import pandas as pd 
import matplotlib.pyplot as pd
%matplotlib inline
import math

## Q4

In [10]:
men=np.array([11,13,24,14,20,34,25,22,49,8])
wom=np.array([9,41,10,21,23,25,37,27,15,7])
n1=len(men)
n2=len(wom)
m_mu=men.mean()
w_mu=wom.mean()
m_std=men.std()
w_std=wom.std()
print(m_std,w_std)

In [21]:
m_std/w_std

1.0585824330250924

As the ratio lies within 0.5-2 so we can safely assume the similarity of variance required to calculate the confidence interval for independent variables. 

In [18]:
N=(n1-1)*m_std*m_std + (n2-1)*w_std*w_std
D=n1+n2-2
Sp=np.sqrt(N/D)
SE = Sp*np.sqrt(1/n1 + 1/n2)
l_95 = [(m_mu -w_mu) - 2.10092*SE,(m_mu-w_mu) + 2.10092*SE]
l_99 = [(m_mu-w_mu) - 2.87844*SE,(m_mu-w_mu) + 2.87844*SE]

In [24]:
print(l_95,l_99)
print(m_mu-w_mu)
print(SE)

[-10.126793123757704, 11.126793123757704] [-14.05961502539322, 15.05961502539322]
0.5
5.058161721416191


Our best estimate of the difference, the point estimate, is 0.5lpa . The standard error of the difference is 5.058 lpa. We are 95% confident that the difference in mean difference in salary of men and women is between -10 and 11 lpa. In this sample, the men have higher mean salary than women by 0.5lpa. Based on this interval, we also conclude that there is no statistically significant difference in mean salaries between men and women, because the 95% confidence interval includes the null value, zero. Again, the confidence interval is a range of likely values for the difference in means. Since the interval contains zero (no difference), we do not have sufficient evidence to conclude that there is a difference.